test/unit/loader/index.spec.js

```diff
@@ -197,6 +218,79 @@ describe('lib/loader', function () {
       }, 1000);
     });
 
+    it('loads same script once if forceLoad flag not set', function (done) {
+      var timeout;
+      var scriptExecuted = 0;
+
+      global.libLoaderTestCallback = function () {
+        scriptExecuted = 1;
+      };
+
+      loader.loadScript('/base/test/fixtures/lib.loader.loadscript.js', function () {
+        expect(scriptExecuted).to.equal(1);
+
+        global.libLoaderTestCallback = function () {
+          scriptExecuted = 2;
+        };
+
+        loader.loadScript('/base/test/fixtures/lib.loader.loadscript.js', function () {
+          clearTimeout(timeout);
+          expect(scriptExecuted).to.equal(1);
+          done();
+        });
+      });
+
+      timeout = setTimeout(function () {
+        done(new Error('script load timed out'));
+      }, 2000);
+    });
+
+    it('loads same script multiple times if forceLoad flag set', function (done) {
+      var timeout;
+      var scriptExecuted = 0;
+
+      global.libLoaderTestCallback = function () {
+        scriptExecuted = 1;
+      };
+
+      loader.loadScript('/base/test/fixtures/lib.loader.loadscript.js', function () {
+        expect(scriptExecuted).to.equal(1);
+
+        global.libLoaderTestCallback = function () {
+          scriptExecuted = 2;
+        };
+
+        loader.loadScript('/base/test/fixtures/lib.loader.loadscript.js', {
+          forceLoad: true
+        }, function () {
+          clearTimeout(timeout);
+          expect(scriptExecuted).to.equal(2);
+          done();
+        });
+      });
+
+      timeout = setTimeout(function () {
+        done(new Error('script load timed out'));
+      }, 2000);
+    });
+
+    it('loads same script in specified namespaceName', function (done) {
+      var scriptExecuted = 0;
+      var url = '/base/test/fixtures/lib.loader.loadscript.js';
+
+      global.libLoaderTestCallback = function () {
+        scriptExecuted = 1;
+      };
+
+      loader.loadScript(url, {
+        namespaceName: TESTING_NAMESPACE_NAME
+      }, function () {
+        expect(scriptExecuted).to.equal(1);
+        expect(namespace.loadedUrls[url]).to.equal(true);
+        done();
+      });
```

Comment:
This function should attempt to load the same url a second time and assert that it is the same as the loaded script from the first execution of `loadScript`.